src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Logging/ProjectLoggerExtensions.ProjectLoggerBatch.cs

```diff
@@ -43,41 +43,43 @@ public void WriteLine(in StringFormat format)
             {
                 if (IsEnabled)
                 {
-                    Init();
-                    WriteNewLineIfNeeded();
-                    WriteIndent();
-                    Write(format);
+                    StringBuilder builder = GetOrCreateBuilder();
```

Comment:
This code was fine, but this pattern allowed the compiler to track things easier.